libvirt/README.md

```diff
@@ -44,3 +44,222 @@ If you want to set default password for podvm debugging then you can use guestfi
 
 Refer to the [cloud-api-adaptor deployment instructions](../install/README.md#deploy-cloud-api-adaptor)
 * If your libvirt host is not configured to be passwordless make sure you set ssh-key-secret in the [kustomization file](../install/overlays/libvirt/kustomization.yaml)
+
+# Creating an end-to-end environment for testing and development
+
+In this section you will learn how to setup an environment in your local machine to run peer pods with
+the libvirt cloud API adaptor. Bear in mind that many different tools can be used to setup the environment
+and here we just make suggestions of tools that seems used by most of the peer pods developers.
+
+## Requirements
+
+You must have a Linux/KVM system with libvirt installed and the following tools:
+
+- docker (or podman-docker)
+- [kubectl](https://kubernetes.io/docs/reference/kubectl/)
+- [kcli](https://kcli.readthedocs.io/en/latest/)
+
+Assume that you have a 'default' network and storage pools created in libvirtd system instance (`qemu:///system`). However,
+if you have a different pool name then the scripts should be able to handle it properly.
+
+## Create the Kubernetes cluster
+
+Use the [`kcli_cluster.sh`](./kcli_cluster.sh) script to create a simple two VMs (one master and one worker) cluster
+with the kcli tool, as:
+
+```
+./kcli_cluster.sh create
+```
+
+With `kcli_cluster.sh` you can configure the libvirt network and storage pools that the cluster VMs will be created, among
+other parameters. Run `./kcli_cluster.sh -h` to see the help for further information.
+
+If everything goes well you will be able to see the cluster running after setting your Kubernetes config with:
+
+`export KUBECONFIG=$HOME/.kcli/clusters/peer-pods/auth/kubeconfig`
+
+For example, shown below:
+
+```
+$ kcli list kube
++-----------+---------+-----------+---------------------------------------+
+|  Cluster  |   Type  |    Plan   |                  Vms                  |
++-----------+---------+-----------+---------------------------------------+
+| peer-pods | generic | peer-pods | peer-pods-master-0,peer-pods-worker-0 |
++-----------+---------+-----------+---------------------------------------+
+$ kubectl get nodes
+NAME                 STATUS   ROLES                  AGE     VERSION
+peer-pods-master-0   Ready    control-plane,master   6m8s    v1.25.3
+peer-pods-worker-0   Ready    worker                 2m47s   v1.25.3
+```
+
+## Prepare the Pod VM volume
```

Comment:
Can we move this section up to the top please. At the moment there is a far bit of duplication between this and the **Creation of base storage volume** section and if you are like me then we will have already spent far too long finding and installing the correct version of the pre-reqs and building the podvm image before you see the option to do it in the container.

Reply:
Talking about the podvm build in container, I opened an issue https://github.com/confidential-containers/cloud-api-adaptor/issues/550, it should be documented. I don't know if @bpradipt has already written about it somewhere, otherwise I can assign this task to myself.

I general i think libvirt's README should be re-write. So if you don't mind to merge this PR without major changes on it, then I can address the re-write in another PR.

Reply:
Yes - I think that's fine given that you've created the follow on issue for it. Thanks